python/cudf/cudf/core/column/column.py

```diff
@@ -2026,7 +2026,45 @@ def as_column(
         mask = bools_to_mask(as_column(mask).unary_operator("not"))
 
         data = data.set_mask(mask)
+    elif (
+        isinstance(arbitrary, list)
+        and len(arbitrary) > 0
+        and cudf.utils.dtypes.is_column_like(arbitrary[0])
```

Comment:
What if the 0th element is None/NA?

Reply:
Would also break this logic out into another utility

Reply:
I agree with Ashwin. Judging from the `_create_list_column_from_sequences_list` It seems like we are assuming that `arbitrary` can have nulls, but it cannot be the first element.  I could be misunderstanding, but I definitly think we need to handle the case where nulls are the first element (or require that none of the elements are null).

Reply:
Yup, made changes to probe all elements of the list.